src/clone-node.ts

```diff
@@ -102,9 +122,11 @@ async function cloneChildren<T extends HTMLElement>(
     children = toArray<T>((nativeNode.shadowRoot ?? nativeNode).childNodes)
   }
 
-  // Keep only visible elements
+  // Keep only visible elements and delete source tags
   children = children.filter(
-    (child) => child.style == null || child.style.display !== 'none',
+    (child) =>
+      (child.style == null || child.style.display !== 'none') &&
+      (child.tagName == null || child.tagName.toLowerCase() !== 'source'),
```

Comment:
Why deleting source tags ? What happen to a video with source tags ?

Reply:
The library relies on the video tag to generate the snapshot, not on the source tags. For the moment, I don't know why the source tag is causing a problem and preventing the image from displaying correctly.